core/include/detray/definitions/units.hpp

```diff
@@ -23,14 +23,24 @@ struct unit {
     static constexpr scalar_t m{static_cast<scalar_t>(1e3)};
     /// @}
 
+    /// Area, native unit mm2
+    /// @{
+    /// mm²
+    static constexpr scalar_t mm2{static_cast<scalar_t>(1.0)};
+    /// cm²
+    static constexpr scalar_t cm2{static_cast<scalar_t>(1e2)};
+    /// m²
+    static constexpr scalar_t m2{static_cast<scalar_t>(1e6)};
+    /// @}
```

Comment:
It would be safer if you define them using the mm, cm and m
For example,
`static constexpr scalar_t mm2{mm * mm};`

Reply:
Maybe, but then it also has the tiny rounding error from the multiplication. Similarly, I also did not use single precision literals